### PR TITLE
Fix start date calculation

### PR DESCRIPTION
**Pull request by katzmo**

### CODE REVIEW

File: src/Utils.php [hunk 1]
```diff
@@ -137,28 +137,46 @@ public static function subscriptionData(\PaymentLineItem $line_item) {
...
-    $count = $recurrence->interval_value ?? 1;
+    $unit = 'month';
+    $count = 1;
+    if (!empty($recurrence->month)) {
+      $interval_unit = $recurrence->interval_unit;
+      if ($interval_unit == 'yearly') {
+        $unit = 'year';
+      }
+      elseif ($interval_unit == 'monthly' && !empty($interval_value = $recurrence->interval_value)) {
+        $count = 12 % $interval_value == 0 ? $interval_value : $count;
```
Comment (torotil) [inline src/Utils.php]: It would be better to explicitly have `1` as else-value. We don’t want to keep the value (whatever it is) we want to set it to `1`.

**Review by torotil — COMMENTED** [review]
Perfect except for the comment and that’s debatable.

File: src/Utils.php (hunk 1 above)
Comment (katzmo) [reply src/Utils.php]: Re-wrote the whole if statement, there was another glitch in it too (weekly payments without month). I hope it’s clearer now!

**Comment by katzmo** [comment]
We’re clearly missing a _day of the week_ field for recurring payments, this would make the start date calculation even more interesting! :stuck_out_tongue_closed_eyes:

File: src/Utils.php [hunk 2]
```diff
@@ -137,28 +137,49 @@ public static function subscriptionData(\PaymentLineItem $line_item) {
...
+        if (!empty($recurrence->month) && !empty($interval_value = $recurrence->interval_value) && 12 % $interval_value == 0) {
+          $count = $interval_value;
+        }
+        break;
+      case 'yearly':
+        if (!empty($recurrence->month)) {
+          $unit = 'year';
+        }
+        break;
+    }
```
Comment (torotil) [inline src/Utils.php:186]: I think it’s still a bit tricky to intepret. How about:

```suggestion
    $find_increment = function ($recurrence) {
      switch ($recurrence->interval_unit) {
        case 'weekly':
          return '1 day';
        case 'monthly':
          if (!empty($recurrence->month) && !empty($value = $recurrence->interval_value) && 12 % $value == 0) {
            return "$value month";
          }
          return '1 month';
        case 'yearly':
          return !empty($recurrence->month) ? '1 year' : '1 month';
      }
    };
    $increment = $find_increment($recurrence); 
```

… or even put this into a protected function. (Sorry I don’t know how to make suggestions for multiple lines at once.

This way each branch of the switch can be read on its own without contextual knowledge of the variables’ prior values.

Comment (katzmo) [reply src/Utils.php:186]: You’re right, that’s more explicit, thanks!

**Review by torotil — APPROVED** [review]

**Pull request merged**